server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -373,4 +482,117 @@ mod tests {
+            .await
+            .unwrap();
+
+        let receiver_1 = TestReceiver::start(axum::http::StatusCode::OK);
+        let receiver_2 = TestReceiver::start(axum::http::StatusCode::OK);
+
+        // Wait at most a second (50ms * 20 retries) for a successful response from that endpoint
+        run_with_retries(
+            || {
+                let endp_1 = receiver_1.endpoint.clone();
+                let endp_2 = receiver_2.endpoint.clone();
+                async {
+                    let client = reqwest::Client::new();
+
+                    if client
+                        .post(endp_1)
+                        .json(&serde_json::json!({"test": "value"}))
+                        .send()
+                        .await?
+                        .status()
+                        != StatusCode::OK
+                    {
+                        anyhow::bail!("Status not OK")
+                    };
+
+                    if client
+                        .post(endp_2)
+                        .json(&serde_json::json!({"test": "value"}))
+                        .send()
+                        .await?
+                        .status()
+                        != StatusCode::OK
+                    {
+                        anyhow::bail!("Status not OK")
+                    };
+
+                    Ok(())
+                }
+            },
+            20,
+        )
+        .await
+        .unwrap();
+
+        let endp_id_1 = create_test_endpoint(&client, &app_id, &receiver_1.endpoint)
+            .await
+            .unwrap();
+        let endp_id_2 = create_test_endpoint(&client, &app_id, &receiver_2.endpoint)
+            .await
+            .unwrap();
+
+        let msg_1 = create_test_message(&client, &app_id, serde_json::json!({"test": "data1"}))
+            .await
+            .unwrap();
+        let msg_2 = create_test_message(&client, &app_id, serde_json::json!({"test": "data2"}))
+            .await
+            .unwrap();
+        let msg_3 = create_test_message(&client, &app_id, serde_json::json!({"test": "data3"}))
+            .await
+            .unwrap();
+
+        // And wait at most two seconds for all attempts to be processed
+        run_with_retries(
+            || async {
+                let list_1: ListResponse<MessageAttemptOut> = client
+                    .get(
+                        &format!("api/v1/app/{}/attempt/endpoint/{}/", app_id, endp_id_1),
+                        StatusCode::OK,
+                    )
+                    .await
+                    .unwrap();
+                let list_2: ListResponse<MessageAttemptOut> = client
+                    .get(
+                        &format!("api/v1/app/{}/attempt/endpoint/{}/", app_id, endp_id_2),
+                        StatusCode::OK,
+                    )
+                    .await
+                    .unwrap();
+
+                for (num, list) in [list_1, list_2].into_iter().enumerate() {
```

Comment:
This shouldn't be inside run with retries
This function is just a "sync gate" to make sure we got everything